agreement_sale_subscription_suspension/models/sale_subscription.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (C) 2019 - TODAY, Open Source Integrators
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import models
+
+
+class SaleSubscriptionSuspension(models.Model):
+    _inherit = 'sale.subscription'
+
+    def suspend_service_profile(self):
+        sp_ids = self.env['agreement.serviceprofile'].search([
+            ('agreement_id', '=', self.agreement_id.id)])
```

Comment:
Chances of `Expected Singleton `as `suspend_service_profile` can be called on multiple browse records.
(Same for `activate_service_profile` method)

Reply:
@Swapnesh-SerpentCS would `@api.multi` resolve this when placed before the function?

Reply:
> @Swapnesh-SerpentCS would `@api.multi` resolve this when placed before the function?

No, You should iterate the loop use one recordset instead of all recordsets (`self`)

Something like this
```
def suspend_service_profile(self):
    for each_service_profile in self:
       self.env['agreement.serviceprofile'].search([
            ('agreement_id', '=', each_service_profile.agreement_id.id)]).write({'state':'suspended'})
```

Reply:
I took a couple of swings at this. I'm not sure how to do everything correctly in a single for loop. It was my understanding that because write() can be applied to multiple records at once, the `Expected Singleton` might not be an issue?

Reply:
@Swapnesh-SerpentCS The methods activate_service_profile() and suspend_service_profile() can only be called from the form view by clicking the "suspend" button or the "suspend" state on a sale_subscription. Because it is always called from the form view, self will always be one record. Can you give me an example of self returning multiple records in the context of these two methods?

Reply:
I believe that @osi-scampbell  and I resolved this.